File: Week 6/sentimental-credit/credit.py
import sys

evens = 0
odds = 0
tmp = 0
start = 0

# get card in´puy
card = int(input("Number: "))

# create 2 copies
very_start = card
start = card

# If 1 number still without calculate
while card > 0:

    # calculate even position numbers
    evens += card % 10
    card = card // 10

    tmp = card % 10 * 2

    # check if the number has 2 digits
    if tmp > 9:
        odds += tmp % 10
        tmp = tmp//10
        odds += tmp % 10
        card = card // 10
    else:
        odds += tmp
        card = card // 10

# caluculate check
check = evens + odds

# calculate the len of the card
tmp = len(str(very_start))

for i in range(tmp - 2):
    start = start // 10

# check if is a real card
if ((check % 10) != 0):
    print("INVALID")
    sys.exit(1)

elif start == 34 or start == 37:
    print("AMEX")
    sys.exit(0)

elif start >= 51 and start <= 55:
    print("MASTERCARD")
    sys.exit(0)

elif start // 10 == 4 and tmp >= 13 and tmp <= 16:
    print("VISA")
    sys.exit(0)
    
else:
    print("INVALID")
    sys.exit(1)

File: Week 1/credit/credit.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    long start_card = 0;
    long card = 0;
    int odds = 0;
    int evens = 0;
    int check = 0;
    int helper = 0;
    int len = 0;

    card = get_long("Number: ");

    start_card = card;

    //check the card
    for (int i = 0; i < 16; i++)
    {
        odds += card % 10;
        card /= 10;

        if ((card % 10 * 2) > 9)
        {
            // if the multiplication of the number has 2 digits, separate
            helper = (card % 10 * 2);
            // add one digit
            evens += helper % 10;
            helper /= 10;
            // add second digit
            evens += helper;
        }
        else
        {
            evens += (card % 10 * 2);
        }

        card /= 10;

    }
    /*
    printf("%i\n", odds);
    printf("%i\n", evens);
    */

    check = evens + odds;

    //check if the card is valid
    if (check % 10 != 0)
    {
        printf("INVALID\n");
        return 0;
    }

    // check for the card leght
    if (start_card >= 1000000000000)
    {
        len = 13;
    }
    if (start_card >= 10000000000000)
    {
        len = 14;
    }
    if (start_card >= 100000000000000)
    {
        len = 15;
    }
    if (start_card >= 1000000000000000)
    {
        len = 16;
    }

    // check for the starting digits
    for (int i = 0; i < (len - 2); i++)
    {
        start_card /= 10;
    }

    //check for the card type
    if (start_card == 34 || start_card == 37)
    {
        printf("AMEX\n");
        return 0;
    }
    else if (start_card >= 51 && start_card <= 55)
    {
        printf("MASTERCARD\n");
        return 0;
    }
    else if ((start_card /= 10) == 4)
    {
        printf("VISA\n");
        return 0;
    }
    else
    {
        printf("INVALID\n");
        return 0;
    }

}